crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -125,6 +125,92 @@ impl Ellipse {
     }
 }
 
+/// A primitive shape formed by the region between two circles, also known as a ring.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Ring")]
+pub struct Annulus {
+    /// The inner circle of the annulus
+    pub inner_circle: Circle,
+    /// The outer circle of the annulus
+    pub outer_circle: Circle,
+}
+impl Primitive2d for Annulus {}
+
+impl Default for Annulus {
+    /// Returns the default [`Annulus`] with radii of `0.5` and `1.0`.
+    fn default() -> Self {
+        Self {
+            inner_circle: Circle { radius: 0.5 },
+            outer_circle: Circle { radius: 1.0 },
+        }
+    }
+}
+
+impl Annulus {
+    /// Create a new [`Annulus`] from an inner and outer circle
+    #[inline(always)]
+    pub const fn new(inner_circle: Circle, outer_circle: Circle) -> Self {
+        Self {
+            inner_circle,
+            outer_circle,
+        }
+    }
+
+    /// Get the diameter of the annulus
+    #[inline(always)]
+    pub fn diameter(&self) -> f32 {
+        self.outer_circle.diameter()
+    }
+
+    /// Get the thickness of the annulus
+    #[inline(always)]
+    pub fn thickness(&self) -> f32 {
+        self.outer_circle.radius - self.inner_circle.radius
+    }
+
+    /// Get the area of the annulus
+    #[inline(always)]
+    pub fn area(&self) -> f32 {
+        self.outer_circle.area() - self.inner_circle.area()
```

Comment:
Nit: Unless rustc optimizes this, there's an extra multiplication by PI here compared to the previous `pi * (outer_radius.powi(2) - inner_radius.powi(2))`.

```suggestion
        PI * (self.outer_circle.radius.powi(2) - self.inner_circle.radius.powi(2))
```

Reply:
This is probably also why the area value changed, which indicates that the compiler does *not* optimize this, at least for tests